common/testing/taskpoller/taskpoller.go

```diff
@@ -46,30 +47,41 @@ type (
 		client    workflowservice.WorkflowServiceClient
 		namespace string
 	}
-	Options struct {
-		tv                  *testvars.TestVars
-		timeout             time.Duration
-		pollStickyTaskQueue bool
+	workflowTaskPoller struct {
+		*TaskPoller
+		pollWorkflowTaskRequest *workflowservice.PollWorkflowTaskQueueRequest
 	}
-	// OptionFunc is a function to change an Options instance
-	OptionFunc func(*Options)
+	activityTaskPoller struct {
+		*TaskPoller
+		pollActivityTaskRequest *workflowservice.PollActivityTaskQueueRequest
+	}
+	options struct {
+		tv      *testvars.TestVars
+		timeout time.Duration
+	}
+	optionFunc func(*options)
 )
 
 var (
-	// DrainWorkflowTask returns an empty RespondWorkflowTaskCompletedRequest response
+	// DrainWorkflowTask returns an empty RespondWorkflowTaskCompletedRequest
 	DrainWorkflowTask = func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondWorkflowTaskCompletedRequest, error) {
 		return &workflowservice.RespondWorkflowTaskCompletedRequest{}, nil
 	}
+	// CompleteActivityTask returns a RespondActivityTaskCompletedRequest with an auto-generated `Result` from `tv.Any().Payloads()`.
+	CompleteActivityTask = func(tv *testvars.TestVars) func(task *workflowservice.PollActivityTaskQueueResponse) (*workflowservice.RespondActivityTaskCompletedRequest, error) {
+		return func(task *workflowservice.PollActivityTaskQueueResponse) (*workflowservice.RespondActivityTaskCompletedRequest, error) {
+			return &workflowservice.RespondActivityTaskCompletedRequest{
+				Result: tv.Any().Payloads(),
+			}, nil
+		}
+	}
 	// WithTimeout defines a timeout for a task poller method (includes *all* RPC calls it has to make)
-	WithTimeout = func(timeout time.Duration) OptionFunc {
-		return func(o *Options) {
+	WithTimeout = func(timeout time.Duration) optionFunc {
+		return func(o *options) {
 			o.timeout = timeout
 		}
 	}
-	// WithPollSticky will make the poller use the sticky task queue instead of the normal task queue
-	WithPollSticky OptionFunc = func(o *Options) {
-		o.pollStickyTaskQueue = true
-	}
+	NoTaskAvailable = errors.New("no task available")
```

Comment:
I always hated this message :-). Can you:
1. Separate WFT and AT by explicitly saying what task is not available.
2. Prefix error message with "test task poller:..." or something like this to be explicitly clear that this message is coming from tests.
3. Maybe rephrase entire message to be more explicit: "PollActivityTask API didn't return any task after waiting for X seconds. It means that ActivityTask was never created." I am not sure about exact text, but I want this message to be more actionable for someone who is not familiar with the system.